src/classes/RLLP_OppRollup.cls

```diff
@@ -80,109 +91,199 @@ public without sharing class RLLP_OppRollup {
+
+            if (role.Contact.npo02__Household__c != null) {
+                householdById.put(role.Contact.npo02__Household__c, role.Contact.npo02__Household__r);
             }
-        }       
-            
-        if (cmap.size() > 0){              
-            RLLP_OppRollup oppR = new RLLP_OppRollup();
-            oppR.rollupContacts(cmap);   
-            if (hhmap.size() > 0){
-                oppR.rollupHouseholds(hhmap);    
-            }        
-        }           
+
+        }
+
+        if (contactById.isEmpty()) {
+            return;
+        }
+
+        RLLP_OppRollup oppRollup = new RLLP_OppRollup();
+        oppRollup.rollupContacts(contactById);
+
+        if (!householdById.isEmpty()) {
+            oppRollup.rollupHouseholds(householdById);
+        }
+    }
+
+    /**
+     * @description Calls the appropriate rollup engine (legacy or crlp) for the account ids provided
+     * @param accountIds Ids for the accounts that require rollups to be recalculated
+     */
+    public void rollupAccounts(Set<Id> accountIds) {
+        if (accountIds == null || accountIds.isEmpty()) {
+            return;
+        }
+
+        if (isCustomizableRollupsEnabled) {
+            runCustomizableRollups(accountIds);
+
+            return;
+        }
+
+        if (isImmediate(accountIds)) {
+            rollupAccounts(getAccountById(accountIds));
+
+        } else {
+            RLLP_OppRollup.rollupAccountsFuture(accountIds);
+        }
+    }
+
+    /**
+     * @description Calls the appropriate rollup engine (legacy or crlp) for the contact ids provided
+     * @param contactIds Ids for the contacts that require rollups to be recalculated
+     */
+    public void rollupContacts(Set<Id> contactIds) {
+        if (contactIds == null || contactIds.isEmpty()) {
+            return;
+        }
+
+        if (isCustomizableRollupsEnabled) {
+            runCustomizableRollups(contactIds);
+
+            return;
+        }
+
+        if (isImmediate(contactIds)) {
+            rollupContacts(getContactById(contactIds));
+
+        } else {
+            RLLP_OppRollup.rollupContactsFuture(contactIds);
+        }
+    }
+
+    /**
+     * @description Calls the future or immediate customizable rollups depending on current context
+     * @param  ids The ids of the sObjects that require rollups to be recalculated
+     */
+    private void runCustomizableRollups(Set<Id> ids){
+        if (isImmediate(ids)) {
+            CRLP_RollupQueueable.runRollupsForIds(new List<Id>(ids));
+
+        } else {
+            System.enqueueJob(new CRLP_RollupQueueable(new List<Id>(ids)));
+        }
+    }
+
+    /**
+     * @description Determines if the context is already async or if there is only 1 record to update
+     * @param ids The set of ids that need their rollups refreshed
+     * @return Boolean - true when the rollups should happen immediately
+     */
+    private Boolean isImmediate(Set<Id> ids) {
```

Comment:
How about `canRunImmediate` for this?

Reply:
Going with should since it technically always can :)